pandas/tests/indexes/multi/test_indexing.py

```diff
@@ -396,7 +396,7 @@ def test_get_loc_missing_nan():
         idx.get_loc(3)
     with pytest.raises(KeyError, match=r"^nan$"):
         idx.get_loc(np.nan)
-    with pytest.raises(TypeError, match=r"'\[nan\]' is an invalid key"):
+    with pytest.raises(TypeError, match="unhashable type: 'list'"):
```

Comment:
Will fixing the underlying issue change the error message raised? If so, then let's not include the error message in what we're testing.

Reply:
By "underlying issue" do you mean the reported regression or the deeper problems in MultiIndex.get_loc mentioned in the OP?

I'll see what it takes to avoid changing this message.

Reply:
The deeper problems in MultiIndex.get_loc.

And to clarify: I'm fine with changing the message. I'd just prefer to not test it if it's likely to change again in the future.

Reply:
OK.  Best guess at this point is that "unhashable type" will be the expected error message eventually.